light-client/src/errors.rs

```diff
@@ -57,6 +60,9 @@ pub enum ErrorKind {
 
     #[error("invalid light block: {0}")]
     InvalidLightBlock(#[source] VerificationError),
+
+    #[error("internal channel disconnected")]
+    ChannelDisconnected,
```

Comment:
Wouldn't it be nice to "decorate" the error with the underlying crossbeam error message?

Reply:
The reason we jump through these hoops is that the crossbeam errors don't derive `Deserialize` and `Serialize` which we need for our errors. Furthermore they only carry the message that failed to send, Would question how much value that has.

Reply:
Oh, I thought you can wrap errors in the error kinds which do not derive (de)serialize and still have serialization via anomaly but I might be wrong (or too tired). Doesn't sound like it's worth to further investigate here.